File: linguagem_java/encapsulamento/empresa/Supervisor.java
package encapsulamento.empresa;

public class Supervisor {
    // Atributos da classe Supervisor
    private String nomeDoSupervisor;
    private Empresa empresa;

    // Construtor da classe Supervisor
    Supervisor(String nomeDoSupervisor, Empresa empresa) {
        setNomeDoSupervisor(nomeDoSupervisor);
        setEmpresa(empresa);
    }

    // Getter e setter de nomeDoSupervisor
    public String getNomeDoSupervisor() {
        return nomeDoSupervisor;
    }

    public void setNomeDoSupervisor(String nomeDoSupervisor) {
        this.nomeDoSupervisor = nomeDoSupervisor;
    }

    // Getter e setter de empresa
    public Empresa getEmpresa() {
        return empresa;
    }

    public void setEmpresa(Empresa empresa) {
        this.empresa = empresa;
    }

    // Método que retorna informações
    String informarSobreSupervisor() {
        return "\nO supervisor " + nomeDoSupervisor + " trabalha na empresa " + empresa.getNomeDaEmpresa();
    }
}

File: linguagem_dart/encapsulamento/empresa/supervisor.dart
import 'empresa.dart';

class Supervisor {
  // Atributos da classe Supervisor
  String _nomeDoSupervisor = "";
  late Empresa _empresa;

  // Construtor da classe Supervisor
  Supervisor(String nome, Empresa empresa) {
    setNomeDoSupervisor = nome;
    setEmpresa = empresa;
  }

  // Getter e setter de empresa
  Empresa get empresa => _empresa;

  set setEmpresa(Empresa empresa) {
    _empresa = empresa;
  }

  // Getter e setter de nomeDoSupervisor
  String get getNomeDoSupervisor => _nomeDoSupervisor;

  set setNomeDoSupervisor(String nomeDoSupervisor) {
    _nomeDoSupervisor = nomeDoSupervisor;
  }

  // Método que retorna as informações
  String retornarInformacoesSupervisor() {
    return "\nO supervisor $_nomeDoSupervisor trabalha na empresa ${_empresa.getNomeDaEmpresa}\n";
  }
}
